### ISSUE TITLE
Extension not being activated

### ISSUE THREAD

**Issue by YusufSuleman** [microsoft/vscode-generator-code#413]
Hello, I am making an extension, if I run it, the extension doesn't even get loaded. It doesn't say that the extension is even activated, in the debug console. I have the default code generated by making a new typescript extension, I am using Arch Linux which was updated a few days ago.

**Comment by sonicwang1989** [comment]
modify activationEvents in package.json like this:
"activationEvents": [
    "*"
  ],

**Comment by YusufSuleman** [comment]
@sonicwang1989 Still not working, nothing changed

**Comment by kexposito** [comment]
@YusufSuleman, having the same issue here 👍

**Comment by kexposito** [comment]
@YusufSuleman, I had the same issue, just updated visual code and all it's working fine now 🎉

**Comment by thefish1087** [comment]
In your `package.json` file, your `extension manifest`, check what version of `vs code` you guys are targeting:

```json
    "engines": {
        "vscode": "^1.80.0"
    },
```

The `vscode-generator-code` tool always target the newest `vs code` version, and if your current version is lower, your extension won't start up.

Check your current version by going `Help>about` in the menu var (Top of `vs code` or by pressing alt`.

Happy coding!

**Comment by mikesmullin** [comment]
Yes I found under Help menu there was an option to restart to apply pending update. Then it worked.
I wonder if the project triggered vscode to recognize it had some new dependency, or if its just been that long for me?

**Comment by debemdeboas** [comment]
I'm having the same issue. Tried with JS and TS, no dice. I'm using WSL2, if that makes any difference. The extension doesn't seem to be loaded at all.

**Comment by kexposito** [comment]
Try using this engine: 
```
  "engines": {
    "vscode": "^1.78.0"
  }
```

**Comment by debemdeboas** [comment]
> Try using this engine:
> 
> ```
>   "engines": {
>     "vscode": "^1.78.0"
>   }
> ```

This worked! Thank you! Any idea as to a more permanent fix?

**Comment by kexposito** [comment]
> > Try using this engine:
> > ```
> >   "engines": {
> >     "vscode": "^1.78.0"
> >   }
> > ```
> 
> This worked! Thank you! Any idea as to a more permanent fix?

I have no clue about that :/ I realize that the issue was that version. So I tried change on my [addon](https://github.com/kexposito/focus-it/commit/77f81d38b9ce7ee9154608c1d585c8b84f0090c7)  in it and it worked. Really happy that these also help you

**Comment by Asu-rada** [comment]
> @YusufSuleman modify activationEvents in package.json like this: "activationEvents": [ "*" ],

Thank you for your method. It does work, but I would like to know why doing so can solve the problem?

**Comment by BestBugMaker** [comment]
> Try using this engine:
> 
> ```
>   "engines": {
>     "vscode": "^1.78.0"
>   }
> ```

Can u tell me how to add this config?

**Comment by jtaavola** [comment]
> > Try using this engine:
> > ```
> >   "engines": {
> >     "vscode": "^1.78.0"
> >   }
> > ```
> 
> Can u tell me how to add this config?

It's in the `package.json`. The generator will specify latest version of VSCode, so if you're not up to date the extension won't be activated. You can either chance `engines.vscode` to the VSCode version you are on, or update your VSCode.

**Comment by Narayanbhat166** [comment]
Hey, the issue still seems to persist. However when i changed the engine to the above mentioned one `^1.78.0` it seems to work. Any idea why this is happening?

**Comment by jtaavola** [comment]
> Hey, the issue still seems to persist. However when i changed the engine to the above mentioned one `^1.78.0` it seems to work. Any idea why this is happening?

`engines.vscode` indicates the version of vscode that your extension works on. The generator defaults to the latest version of vscode I believe.

https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines
